### DISCUSSION TITLE
Customization of the built-in standard Infinispan configuration as seen in Wildfly based KC

### DISCUSSION THREAD

**Discussion by psytester** [keycloak/keycloak#10694] [Archive]
Based on KC 17.0.0 and 18.0.0-SNAPSHOT

---

Is there any plan to provide a simple basic customization of the built-in standard Infinispan address and port configuration as seen in WildFly based KC?

In `standalone-ha.xml` we had environment variables that could change the default value.

_grep "\${" /opt/keycloak/standalone/configuration/standalone-ha.xml_
```
  ....

            <core-environment node-identifier="${jboss.tx.node.id:1}">
  ....
            <inet-address value="${jboss.bind.address.management:127.0.0.1}"/>
            <inet-address value="${jboss.bind.address.private:127.0.0.1}"/>
            <inet-address value="${jboss.bind.address:127.0.0.1}"/>
 ....
        <socket-binding name="jgroups-mping" interface="private" multicast-address="${jboss.default.multicast.address:230.0.0.4}" multicast-port="45700"/>
        <socket-binding name="jgroups-udp" interface="private" port="55200" multicast-address="${jboss.default.multicast.address:230.0.0.4}" multicast-port="45688"/>
        <socket-binding name="modcluster" multicast-address="${jboss.modcluster.multicast.address:224.0.1.105}" multicast-port="23364"/>
 ....
```

These values were modified via `$JBOSS_OPTS` or `$JAVA_OPTS`
Huge Benefit: No (manual) modification of standalone configuration files.

Simply setup on node 1:
_JBOSS_OPTS="$JBOSS_OPTS -Djboss.node.name=keycloak-1 -Djboss.tx.node.id=keycloak-1 -Djboss.bind.address.management=IP_of_private_management_NIC -Djboss.bind.address=IP_of_public_NIC -Djboss.default.multicast.address=......."_

Comparing Wildfly and Quarkus Infinispan config, the override of default port values would be great, too!
Something like similar configuration notation:
```
${jgroups.mping.multicast.port:45700}
${jgroups.udp.port:55200}
${jgroups.udp.multicast.port:45688}
.....
```

At the end, I need other ip addresses and ports but I don't want to deploy my own `cache-config-file` as I want to use mostly the build-in Infinispan config

**Comment by csergiu20** [comment]
This would be a great feature, to be able to add a simple configuration for Infinispan directly from keycloak.conf.

**Reply by tacerus** [reply]
Agreed!
